library/src/main/java/ai/promoted/sdk/UpdateClientConfigUseCase.kt

```diff
@@ -0,0 +1,53 @@
+package ai.promoted.sdk
+
+import ai.promoted.ClientConfig
+import ai.promoted.config.RemoteConfig
+import ai.promoted.config.RemoteConfigServiceFinder
+
+/**
+ * The purpose of this class is to take the latest cached [RemoteConfig] and merge it with the
+ * compiled [ClientConfig] that the library user is passing in. The output will be a new
+ * [ClientConfig] that prefers values from the [RemoteConfig] (since it's dynamic) over the original
+ * baseline/compiled/hard-coded [ClientConfig] used to initialize the SDK.
+ *
+ * Any remotely configurable values on the original [ClientConfig] will be overridden by their
+ * [RemoteConfig] counterparts if they are non-null in the latest [RemoteConfig].
+ */
+internal class UpdateClientConfigUseCase(
+    remoteConfigServiceFinder: RemoteConfigServiceFinder
+) {
+    private val service = remoteConfigServiceFinder.findAvailableService()
+
+    /**
+     * Ensure any [RemoteConfig] values are applied on top of this baseline [ClientConfig].
+     */
+    fun updateClientConfig(baselineConfig: ClientConfig): ClientConfig {
+        /* Kick off a remote-config update that, if updated, will apply to the *next* session */
```

Comment:
nit: You're using // style comments everywhere else, why not here?

Reply:
It's kind of a side-effect of the `updateClientConfig` function--the update referred to in the function name is "superimpose that latest remote config values onto my compiled ClientConfig". The fetch function, which should be done at the same point in the app lifecycle (pre-fetching for the _next_ session) is not directly updating the current compiled clientConfig that is coming in. That was just my way of calling out slightly special behavior since it's a pre-fetch for the next session rather than a synchronous fetch/update for the current session.